cumulus_library/template_sql/shared_macros/syntax.sql.jinja

```diff
@@ -17,4 +17,16 @@
 UNION
 {%- endif -%}
 {%- endmacro -%}
+
+{#- The intent of this function is to allow an end user to work in SQL for partial
+string searches, rather than regex. So for example, a value of macro_string like:
```

Comment:
This docstring could use a link to the regex syntax being used. Like the [trino docs](https://trino.io/docs/current/functions/regexp.html) I guess (and/or the java docs it then links to).